superset-frontend/src/explore/components/controls/AnnotationLayerControl/AnnotationLayer.jsx

```diff
@@ -267,38 +190,33 @@ class AnnotationLayer extends React.PureComponent {
         errors.push(validateNonEmpty(intervalEndColumn));
       }
     }
-    errors.push(!this.isValidFormulaAnnotation(value, annotationType));
+    errors.push(!isValidFormulaAnnotationHandler(value, annotationType));
```

Comment:
name,
      annotationType,
      sourceType,
      value,
      timeColumn,
      intervalEndColumn,
      should all be referencing state, but they are referencing props.